src/routers/datatypeRouters/edges/regulatory_regions_genes.ts

```diff
@@ -0,0 +1,74 @@
+import { z } from 'zod'
+import { publicProcedure } from '../../../trpc'
+import { loadSchemaConfig } from '../../genericRouters/genericRouters'
+import { RouterEdges } from '../../genericRouters/routerEdges'
+import { geneFormat, genesQueryFormat } from '../nodes/genes'
+import { paramsFormatType } from '../_helpers'
+import { TRPCError } from '@trpc/server'
+import { regulatoryRegionFormat, regulatoryRegionsQueryFormat } from '../nodes/regulatory_regions'
+import { descriptions } from '../descriptions'
+
+const edgeSources = z.object({
+  source: z.enum([
+    'ENCODE_EpiRaction',
+    'ENCODE-E2G-DNaseOnly',
+    'ENCODE-E2G-Full'
+  ]).optional()
+})
+
+function edgeQuery (input: paramsFormatType): string {
+  const query = []
+
+  if (input.source !== undefined) {
+    query.push(`record.source == '${input.source}'`)
+    delete input.source
+  }
+
+  if (Object.keys(input).filter(item => !['page', 'verbose'].includes(item)).length === 0) {
+    throw new TRPCError({
+      code: 'BAD_REQUEST',
+      message: 'At least one property must be defined.'
+    })
```

Comment:
page and verbose shouldn't be required, they usually go with default values: page: `default(0)` and verbose with default `'False'`